msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/AadInstanceDiscoveryProvider.java

```diff
@@ -62,40 +62,39 @@ static InstanceDiscoveryMetadataEntry getMetadataEntry(URL authorityUrl,
                                                            boolean validateAuthority,
                                                            MsalRequest msalRequest,
                                                            ServiceBundle serviceBundle) {
-        String host = authorityUrl.getHost();
-
-        if (shouldUseRegionalEndpoint(msalRequest)) {
-            //Server side telemetry requires the result from region discovery when any part of the region API is used
-            String detectedRegion = discoverRegion(msalRequest, serviceBundle);
 
-            if (msalRequest.application().azureRegion() != null) {
-                host = getRegionalizedHost(authorityUrl.getHost(), msalRequest.application().azureRegion());
-            }
+        String host = authorityUrl.getHost();
 
-            //If region autodetection is enabled and a specific region not already set,
-            // set the application's region to the discovered region so that future requests can skip the IMDS endpoint call
-            if (null == msalRequest.application().azureRegion() && msalRequest.application().autoDetectRegion()
-                    && null != detectedRegion) {
-                msalRequest.application().azureRegion = detectedRegion;
-            }
-            cacheRegionInstanceMetadata(authorityUrl.getHost(), msalRequest.application().azureRegion());
-            serviceBundle.getServerSideTelemetry().getCurrentRequest().regionOutcome(
-                    determineRegionOutcome(detectedRegion, msalRequest.application().azureRegion(), msalRequest.application().autoDetectRegion()));
+        //If instanceDiscovery flag set to false or instance discovery previously failed, do not do instance discovery
+        if (!msalRequest.application().instanceDiscovery() || instanceDiscoveryFailed) {
+            return InstanceDiscoveryMetadataEntry.builder().
+                    preferredCache(host).
+                    preferredNetwork(host).
+                    aliases(Collections.singleton(host)).
+                    build();
         }
 
-        InstanceDiscoveryMetadataEntry result = cache.get(host);
+        //If there is no cached instance metadata, do instance and region discovery and cache the result
+        if (cache.get(host) == null) {
```

Comment:
All this top level logic needs logging statements, which will help investiagte issues.

Reply:
In the latest commit I added debug logs for all the major if blocks: when instance discovery is disabled, when there's no cached metadata, when region API is used, etc.